superset/db_engine_specs/base.py

```diff
@@ -864,18 +864,24 @@ def get_all_datasource_names(
         all_datasources: List[utils.DatasourceName] = []
         for schema in schemas:
             if datasource_type == "table":
-                all_datasources += database.get_all_table_names_in_schema(
-                    schema=schema,
-                    force=True,
-                    cache=database.table_cache_enabled,
-                    cache_timeout=database.table_cache_timeout,
+                all_datasources.extend(
+                    utils.DatasourceName(*datasource_name)
+                    for datasource_name in database.get_all_table_names_in_schema(
+                        schema=schema,
+                        force=True,
+                        cache=database.table_cache_enabled,
+                        cache_timeout=database.table_cache_timeout,
```

Comment:
what would happen if this value returns from cache prior to this change and it's not a list of strings but rather the DatasourceName object?